src/glyph/parse.rs

```diff
@@ -122,6 +122,12 @@ impl<'names> GlifParser<'names> {
         }
 
         self.glyph.load_object_libs()?;
+
+        let mut seen_codepoints = HashSet::new();
+        let deduplicated_codepoints =
+            self.glyph.codepoints.iter().filter(|c| seen_codepoints.insert(**c)).cloned().collect();
+        self.glyph.codepoints = deduplicated_codepoints;
+
```

Comment:
In general, how many codepoints do we expect a glyph to have? I would expect very few, so it might be simpler to just write this as:

```suggestion
        self.glyph.codepoints.sort_unstable();
        self.glyph.codepoints.dedup();
```

it's possible that this would even be better in the case where the collection might be larger, but my intuition there is not well formed.

Reply:
I expect the list to always be very small.

Reply:
Oh, but: the Unicode code point list is ordered, and the first codepoint is considered the "primary" one in the Python libraries, so we can't just sort them 🤔

Reply:
Okay yea a good reason to prefer this.


The other option I could see is using an [`IndexSet`](https://docs.rs/indexmap/1.9.1/indexmap/set/struct.IndexSet.html) to preserve insertion order but also enforce uniqueness? as use it to always store the codepoints. But maybe not worth taking on a dependency for something so small.


The thing that bothers me (irrationally, it is unlikely to matter in theory) is that with this patch we're now making an additional allocation, per-glyph, on both load and save. It might make sense to at least only do this work if there is > 1 codepoint?

Reply:
Hm! Yes sounds reasonable. I'll look into that.